runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -417,7 +416,9 @@ protected boolean fillReadQueue(final long maxGlobal,
                     log.trace("Fill_Read_Queue[{}] Get Stream Address Map using checkpoint with {} entries",
                             this, context.readCpQueue.size());
 
-                    return true;
+                    // Note: this is a checkpoint, we do not need to verify it is before the trim mark, it actually should be
+                    // cause this is the last address of the trimmed range.
+                    context.setGlobalPointer(context.checkpoint.startAddress);
```

Comment:
Can you please add a comment on why we need to set the global pointer, it is not obvious for someone with little context.

Reply:
This part is going to be reverted as per our discussion.